### PR TITLE
Enable .NET Core startup script to take either source path or published output path as input.

### PR DESCRIPTION
**Pull request by kichalla**
In case source path is supplied, then we by default look for the reserved folder name 'oryx_publish_output'.
If found, this folder will become the app path otherwise the source path would be acting as a published output path

### CODE REVIEW

File: tests/Oryx.Integration.Tests/DotNetCoreEndToEndTests.cs [hunk 1]
```diff
@@ -278,7 +276,134 @@ public async Task CanBuildAndRun_NetCore21WebApp_UsingDefaultPublishOutputDirect
...
+                },
+                async (hostPort) =>
+                {
+                    var data = await _httpClient.GetStringAsync($"http://localhost:{hostPort}/");
+                    Assert.Contains("Hello World! from WebApp1", data);
+                });
+        }
+
+        [Fact]
+        public async Task CanBuildAndRunWebApp_UsingDefaultPublishOutputDirectory_AndOutputIsZipped()
```
Comment (kichalla) [inline tests/Oryx.Integration.Tests/DotNetCoreEndToEndTests.cs]: For reference, example output from build and runtime images:

**Build**
```
Build orchestrated by Microsoft Oryx, https://github.com/Microsoft/Oryx
You can report issues at https://github.com/Microsoft/Oryx/issues

Oryx Version      : 0.2.unspecified, Commit: unspecified
Build Operation ID: |Ad8hbsfqNt4=.d8d09ca6_


Source directory     : /oryxtests/MultiWebAppRepo
Destination directory: /oryxtests/MultiWebAppRepo/oryx_publish_output


Using .NET Core SDK Version: 2.1.700

Restoring packages for 'src/WebApp1/WebApp1.csproj'...

  Restore completed in 220.24 ms for /oryxtests/MultiWebAppRepo/src/Greeter/Greeter.csproj.
  Restore completed in 543.1 ms for /oryxtests/MultiWebAppRepo/src/WebApp1/WebApp1.csproj.

Publishing 'src/WebApp1/WebApp1.csproj' to directory '/tmp/puboutput/publish'...

Microsoft (R) Build Engine version 16.1.76+g14b0a930a7 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 45.27 ms for /oryxtests/MultiWebAppRepo/src/Greeter/Greeter.csproj.
  Restore completed in 50.66 ms for /oryxtests/MultiWebAppRepo/src/WebApp1/WebApp1.csproj.
  Greeter -> /oryxtests/MultiWebAppRepo/src/Greeter/bin/Release/netcoreapp2.1/Greeter.dll
  WebApp1 -> /oryxtests/MultiWebAppRepo/src/WebApp1/bin/Release/netcoreapp2.1/MyWebApp.dll
  WebApp1 -> /tmp/puboutput/publish/

Removing existing manifest file
Creating a manifest file...
Manifest file created.

Done in  sec(s).
```

**Runtime**
```
Oryx Version : 0.2.unspecified, Commit: unspecified

Found publish output directory at '/oryxtests/MultiWebAppRepo/oryx_publish_output'. Using it's contents for running the app.
Build Operation ID: |Ad8hbsfqNt4=.d8d09ca6_
Building the startup script...
Read build manifest file and found output has been zipped, so adding script to extract it...
Writing output script to 'run.sh'

Running the script...
Found 'oryx_output.tar.gz' under '/oryxtests/MultiWebAppRepo/oryx_publish_output'. Extracting it's contents into it...
Done.
Deleting the file 'oryx_output.tar.gz'...
Done.
Finding the startup file name...
Found the startup file 'MyWebApp.dll'
Running the command 'dotnet 'MyWebApp.dll''...

Hosting environment: Production
Content root path: /oryxtests/MultiWebAppRepo/oryx_publish_output
Now listening on: http://[::]:3000
Application started. Press Ctrl+C to shut down.
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 GET http://localhost:33051/
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
      Request finished in 18.0374ms 200
```

File: src/startupscriptgenerator/dotnetcore/main.go [hunk 2]
```diff
@@ -44,6 +44,16 @@ func main() {
 		fullAppPath = common.GetValidatedFullPath(*appPathPtr)
 	}
 
+	// If a user supplies the repo root (instead of output folder), we try to see if the repo root has
+	// the default oryx publish output folder
+	defaultPublishOutputDirPath := filepath.Join(fullAppPath, "oryx_publish_output")
```
Comment (dorfire) [inline src/startupscriptgenerator/dotnetcore/main.go]: consolidate `oryx_publish_output` to a shared constant

Comment (dorfire) [reply src/startupscriptgenerator/dotnetcore/main.go]: also, this would only work if the user didn't specify a custom output directory, correct?

**Comment by gpcastro** [comment]
@kichalla is this PR still valid?

**Comment by kichalla** [comment]
> @kichalla is this PR still valid?

I think we can close it for now and can revisit later

**Pull request closed without merging**